Systems/WorldGen/Standard/ChunkGen/1.GenTerra/GenTerra.cs

```diff
@@ -443,9 +436,11 @@ private void OnChunkColumnGen(IServerChunk[] chunks, int chunkX, int chunkZ, ITr
                                             int lZ = zN * lerpHor + z;
 
                                             double geoUpheavelTaper = 0;
-                                            if (posY > taperThreshold && disty[0] <- 2)
+                                            double distYHere = GameMath.BiLerp(distY[0], distY[1], distY[2], distY[3], x * lerpDeltaHor, z * lerpDeltaHor);

```

Comment:
Go ahead and interpolate `distY` to use for the check instead of the interpolation base value `distY[0]`, to avoid potential flat square artifacts.